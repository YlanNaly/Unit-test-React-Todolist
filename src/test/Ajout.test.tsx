import React from 'react';
import {fireEvent, render} from "@testing-library/react";
import AddTask from "../todolist/Ajout";

it("render input add task" , () =>{
    const tree = render(<AddTask addTask={jest.fn} />);
    const input= tree.getByPlaceholderText("Ajouter une tâche");

    expect(input).toMatchSnapshot();
});

test("render function onchange on add task component" , ()=>{
    const value = "test";
    const tree = render(<AddTask addTask={jest.fn()} />);
    const input= tree.getByPlaceholderText("Ajouter une tâche") as  HTMLInputElement;
    fireEvent.change(input, {target: {value :value}})
    expect(input.value).toEqual("test");
})