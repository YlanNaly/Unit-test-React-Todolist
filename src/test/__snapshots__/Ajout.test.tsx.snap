// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render input add task 1`] = `
<input
  class="input-group-text"
  data-testid="toggle"
  placeholder="Ajouter une tâche"
  type="text"
  value=""
/>
`;

exports[`test add task for input 1`] = `
<input
  class="input-group-text"
  data-testid="toggle"
  placeholder="Ajouter une tâche"
  type="text"
  value=""
/>
`;

exports[`test input add Task 1`] = `"titre"`;
